### PR TITLE
Improved users api

### PR DESCRIPTION
**Pull request by qqmbr4k**
https://app.asana.com/0/56970054049737/420670860269594

### CODE REVIEW

File: rest-server/auth-server/src/main/scala/com/actionml/authserver/dal/mongo/UsersDaoImpl.scala [hunk 1]
```diff
@@ -12,25 +13,48 @@ class UsersDaoImpl(implicit inj: Injector) extends UsersDao with MongoSupport wi
...
+  override def find(id: String, secretHash: String)(implicit log: LoggingAdapter): Future[Option[UserAccount]] = {
     users.find(and(
       equal("id", id),
       equal("secretHash", secretHash)
     )).toFuture
-      .recover { case _ => List.empty }
-      .map(_.headOption)
+      .recover { case e =>
+        log.error(s"Can't find user with id $id and password hash $secretHash")
+        List.empty
```
Comment (pferrel) [inline rest-server/auth-server/src/main/scala/com/actionml/authserver/dal/mongo/UsersDaoImpl.scala]: Are you using a different method for logging than Harness? We should use the same if possible and also naming conventions. In Harness we use the com.typesafe.scalalogging.LazyLogging in some places and akka.event.Logging in other places but never akka.event.LoggingAdapter

Can we consolidate these? It's not required now but seems to be a point of divergence though I don't understand the reasons for all these variants of loggers.

File: rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala [hunk 2]
```diff
@@ -31,48 +34,80 @@ import com.actionml.circe.CirceSupport
...
 
-  def route: Route = (handleExceptions(exceptionHandler) & extractLog) { implicit log =>
-    (pathPrefix("auth" / "users") & extractAccessToken) { implicit token =>
-      (pathEndOrSingleSlash & post & hasAccess(user.create) & entity(as[CreateUserRequest])) {
-        case CreateUserRequest(roleSetId, resourceId) =>
-          onSuccess(usersService.create(roleSetId, resourceId))(complete(_))
+  def route: Route = extractLog { implicit log =>
+    (pathPrefix("auth" / "users") & handleExceptions(exceptionHandler(log)) & extractAccessToken) { implicit token =>
+      pathEndOrSingleSlash {
+        (get & parameters('offset.as[Int] ? 0, 'limit.as[Int] ? Int.MaxValue) & hasAccess(user.permissions)) { (offset, limit) =>
```
Comment (pferrel) [inline rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala:46]: Can you document offset/limit in the rest_spec.md?

File: rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala [hunk 3]
```diff
@@ -31,48 +34,80 @@ import com.actionml.circe.CirceSupport
...
+            onSuccess(usersService.grantPermissions(userId, req.roleSetId, resourceId)) {
+              complete(PermissionsResponse(userId, req.roleSetId, resourceId))
+            }
+          } ~
+          (delete & parameters('roleSetId, 'resourceId ?)) { (roleSetId, resourceIdOpt) =>
+            val resourceId = resourceIdOpt.getOrElse(ResourceId.*)
+            onSuccess(usersService.revokePermissions(userId, roleSetId, resourceId)) {
+              complete(PermissionsResponse(userId, roleSetId, resourceId))
+            }
           }
```
Comment (pferrel) [inline rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala:71]: Please document REST API changes in rest_spec.md

**Review by pferrel — COMMENTED** [review]
Looks fine, see questions. Please update the rest_spec.md with all changes, it will be easier than reading the code.

I need that before merging

File: rest-server/auth-server/src/main/scala/com/actionml/authserver/dal/mongo/UsersDaoImpl.scala (hunk 1 above)
Comment (andreyshikov) [reply rest-server/auth-server/src/main/scala/com/actionml/authserver/dal/mongo/UsersDaoImpl.scala]: migrated to LazyLogging

File: rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala (hunk 3 above)
Comment (andreyshikov) [reply rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala:71]: I added offset and limit (default is 0 and MAX_INT), and all other changes were made to fit rest_spec

**Comment by pferrel** [comment]
looks good

**Pull request merged**